doc/citing.md

```diff
@@ -0,0 +1,16 @@
+# Citing QGreenland
```

Comment:
Extracting some content to individual files made it easier to conditionally include it